### ISSUE TITLE
[Bug]: bug title: Proposal to Adjust Default n_steps for PPO for Better Outcomes with Small Environments

### ISSUE THREAD

**Issue by Ian-Sy-Zhang** [DLR-RM/stable-baselines3#1846]
### 🐛 Bug

Hello Stable-Baselines3 Team,

I recently encountered an issue while training an agent on the FrozenLake environment using PPO's default settings. The agent showed no signs of learning, which prompted me to delve into the `stable_baselines3/ppo/ppo.py` source code. I discovered that the default `n_steps` is set to 2048. This high value means that in a single environment setup, the PPO model requires over 2048 steps worth of data before starting to train.

As a beginner, I find Stable-baselines3 incredibly user-friendly. However, I generally only adjusted the batch_size and learning_rate, not being aware of the impact of `n_steps`. This oversight led to some unsuccessful training attempts with small sample sizes using the default PPO parameters.

My breakthrough came when I reduced `n_steps` to 16, which significantly improved the training results. This experience led me to look at similar parameters in other algorithms, like DQN. I noticed that in merge request #1785, the `learning_starts` parameter for `dqn.py` was initially 50000, which was later reduced to 100 after recognizing that it was too high compared to other algorithms. This change aligns with the improved training outcomes I observed using the default settings of the newer DQN model on my small sample data.

Considering these observations, I suggest revising the default `n_steps` value in `ppo.py` from 2048 to a smaller number. I believe this could be more beginner-friendly and may enhance the usability of Stable-baselines3, particularly for those working with smaller environments.

Thank you for considering this adjustment.

Best regards,
Shiyu

### To Reproduce

import bug_lib as BL
import subprocess
import os
import sys

sys.path.insert(0, './training_scripts/')
from stable_baselines3 import DQN, PPO, A2C, SAC
from stable_baselines3.common.callbacks import BaseCallback

import training_scripts.Env

def get_PPO_Model(env, model_path=os.path.join('RLTesting', 'logs', 'ppo.zip')):
    if os.path.isfile(model_path):
        print("loading existing model")
        model = PPO.load(model_path, env=env)
    else:
        print("creating new model")
        model = PPO('MlpPolicy', env)
        # new_logger = configure(folder="logs", format_strings=["stdout", "log", "csv", "tensorboard"])
        # model.set_logger(new_logger)
    return model

'''
note here: max_steps=200 so that it's less than the default value of 2048; maybe making 2048 a small value will be better
'''
def train_PPO_model(model, max_steps=200, model_path=os.path.join('RLTesting', 'logs', 'ppo.zip')):
    vec_env = model.get_env()
    vec_env.reset()
    vec_env.render()
    model.learn(max_steps)
    action_state_list = vec_env.envs[0].get_state_action_pairs()
    model.save(model_path)
    vec_env.close()
    return action_state_list

if __name__ == '__main__':
    env = training_scripts.Env.EnvWrapper()
    env.reset()
    model = get_PPO_Model(env=env)
    result = train_PPO_model(model=model)

### Relevant log output / Error message

```shell
max_steps=200, so that it's less than the default value of 2048; The model is untrained. Maybe making 2048 a small value will be better
```

### Checklist

- [X] My issue does not relate to a custom gym environment. (Use the custom gym env template instead)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by Ian-Sy-Zhang** [comment]
sorry about the format of  "To Reproduce", I will rewrite here:

```python
import bug_lib as BL
import subprocess
import os
import sys

sys.path.insert(0, './training_scripts/')
from stable_baselines3 import DQN, PPO, A2C, SAC
from stable_baselines3.common.callbacks import BaseCallback

import training_scripts.Env


def get_PPO_Model(env, model_path=os.path.join('RLTesting', 'logs', 'ppo.zip')):
    if os.path.isfile(model_path):
        print("loading existing model")
        model = PPO.load(model_path, env=env)
    else:
        print("creating new model")
        model = PPO('MlpPolicy', env)
        # new_logger = configure(folder="logs", format_strings=["stdout", "log", "csv", "tensorboard"])
        # model.set_logger(new_logger)
    return model


'''
note here: max_steps=200 so that it's less than the default value of 2048; maybe making 2048 a small value will be better
'''
def train_PPO_model(model, max_steps=200, model_path=os.path.join('RLTesting', 'logs', 'ppo.zip')):
    vec_env = model.get_env()
    vec_env.reset()
    vec_env.render()
    model.learn(max_steps)
    action_state_list = vec_env.envs[0].get_state_action_pairs()
    model.save(model_path)
    vec_env.close()
    return action_state_list


if __name__ == '__main__':
    env = training_scripts.Env.EnvWrapper()
    env.reset()
    model = get_PPO_Model(env=env)
    result = train_PPO_model(model=model)
```

**Comment by Ian-Sy-Zhang** [comment]
Hello:


I still think it's a seminar problem like merge request #1785; You can not assume everyone using Stable-baselines3 is fimilar with every line in your documentation.

I believe that unreasonable parameter settings could lead to users wasting their time, so perhaps adjusting to a smaller parameter or providing prompts such as 'Your model is actually still warming up and not yet training' would be more user-friendly.

Ignore the demo code since every time you call 'model.learn(max_steps)' and max_step < 2048 will cause that problem.


Regards

**Comment by Ian-Sy-Zhang** [comment]
To be percise, check #1760

**Comment by Ian-Sy-Zhang** [comment]
I trained the ppo model for 300 rounds (each round has 1-80 steps), the pic below is using n_steps=16;
![output](https://github.com/DLR-RM/stable-baselines3/assets/45763229/ee22fe65-8c28-49c6-b1c7-87407f557d45)

X axis means round index, and Y axis means accuracy;
Red line means the result of linear regression of the training.


The following picture uses default param, n=2048. The picture shows that the agent didn't learn anything.
![output1](https://github.com/DLR-RM/stable-baselines3/assets/45763229/0f7b6c7a-7393-402f-8f3a-435e5afcfb84)

I think a user-friendly hint 'your model is only warming up' or 'your model is not trained' will help; Or you can just set n_steps a smaller number like the situation #1760
